### PR TITLE
Bug 2033751: Rebase 1.23.0

### PR DESCRIPTION
**Pull request by damemi**
Depends on https://github.com/openshift/kubernetes/pull/1087

### CODE REVIEW

**Comment by vrutkovs** [comment]
/hold

A lot of CSI tests are failing due to kube-apiserver still being 1.22.1

**Comment by jsafrane** [comment]
/test e2e-gcp-csi 
/test e2e-aws-csi

Trying with new CI images.

**Comment by vrutkovs** [comment]
Rebase has landed, lets see if this brings new kapi/kubelet
/retest

**Comment by vrutkovs** [comment]
kube-apiserver is now 1.23 (kubelet is still 1.22.1 from 6859754). All test failures in e2e-gcp are timeouts

/hold cancel

**Comment by jsafrane** [comment]
Lot of the timeouts of storage tests are caused by an Ephemeral container not getting started.

```
STEP: Repeating the test on an ephemeral container (if enabled)
...
fail [k8s.io/kubernetes@v1.23.0/test/e2e/framework/pods.go:173]: Unexpected error:
    <*errors.errorString | 0xc00033ec30>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
occurred
```

Test code: https://github.com/kubernetes/kubernetes/blob/0669da445fa8c1ae07c15c0827f0e83da11cbe58/test/e2e/framework/volume/fixtures.go#L605

Do we have Ephemeral containers enabled?

**Comment by jsafrane** [comment]
I can see that EphemeralContainers were [enabled as beta in 1.23](https://github.com/kubernetes/kubernetes/blob/0669da445fa8c1ae07c15c0827f0e83da11cbe58/pkg/features/kube_features.go#L95-L100), so we may have API server + KCM that has the feature enabled and kubelet that has it disabled -> test failure.

**Comment by vrutkovs** [comment]
/hold

let it pick up a new nightly for kubelet update

**Comment by deads2k** [comment]
force accepted a payload

/test all

**Comment by vrutkovs** [comment]
CSI tests fail due to missing busybox image
```
Back-off pulling image "quay.io/openshift/community-e2e-images:e2e-k8s-gcr-io-e2e-test-images-busybox-1-29-1-4vRlLJlMSKkJ4csR"
```
We only list `1.29-2`:
```
./test/extended/util/image/image.go
67:		"k8s.gcr.io/e2e-test-images/busybox:1.29-2": 7,

./test/extended/util/image/zz_generated.txt
8:k8s.gcr.io/e2e-test-images/busybox:1.29-2 quay.io/openshift/community-e2e-images:e2e-7-k8s-gcr-io-e2e-test-images-busybox-1-29-2-cqcP1Tnbm-JjJyUA
```

**Comment by soltysh** [comment]
> CSI tests fail due to missing busybox image

mirrored

/test e2e-gcp
/test e2e-aws-fips
/test e2e-gcp-upgrade

File: test/extended/util/image/image.go [hunk 1]
```diff
@@ -61,6 +61,8 @@ func init() {
 		"k8s.gcr.io/sig-storage/snapshot-controller:v3.0.2":       -1,
 		"k8s.gcr.io/sig-storage/snapshot-controller:v4.0.0":       -1,
 
+		"k8s.gcr.io/e2e-test-images/busybox:1.29-1": -1,
```
Comment (yselkowitz) [inline test/extended/util/image/image.go:64]: I think this would be easier to maintain if it were a few lines down with busybox:1.29-2

Comment (soltysh) [reply test/extended/util/image/image.go:64]: nah, I prefer to have it next to sig-storage elements, b/c they are handled differently than that other busybox image, it's complicated :sweat:

**Comment by vrutkovs** [comment]
Retesting to include #26745 for better error message

**Comment by soltysh** [comment]
Replaced with https://github.com/openshift/origin/pull/26753
/close

**Pull request closed without merging**